compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -2497,7 +2500,8 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
     }
 
     fn point_at_param_definition(&self, err: &mut Diagnostic, param: ty::ParamTy) {
-        let generics = self.tcx.generics_of(self.body_id.owner.to_def_id());
+        let body_hir_id = self.tcx.hir().local_def_id_to_hir_id(self.body_id);
+        let generics = self.tcx.generics_of(body_hir_id.owner.to_def_id());
```

Comment:
```suggestion
        let generics = self.tcx.generics_of(self.body_id);
```
this previous code looks wrong. We shouldn't use the `HirId` owner to get the generics.